learn-typescript-linting.md

```diff
@@ -0,0 +1,503 @@
+- [3 Adding Standard Style Formatter](#3-adding-standard-style-formatter)
+  - [3.1 Installing Standard](#31-installing-standard)
+  - [3.2 Evaluating new errors](#32-evaluating-new-errors)
+  - [3.3 Configuring Standard specific rules](#33-configuring-standard-specific-rules)
+- [4 Adding Prettier Style Formatter](#4-adding-prettier-style-formatter)
+  - [4.1 Installing Prettier](#41-installing-prettier)
+  - [4.2 Executing Prettier](#42-executing-prettier)
+  - [4.3 Configuring Prettier specific rules](#43-configuring-prettier-specific-rules)
+- [5 Additional Resources and Documentation](#5-additional-resources-and-documentation)
+
+## 1 Getting Started
+[`[toc]`](#table-of-contents)
+
+**Note:*** Developers without an existing TypeScript project should start here at section 1; developers with an existing project can skip ahead to section [2 Adding ESLint](#2-adding-eslint). This guide works best if you follow along with the GitHub repository.
+
+View the GitHub repository [`learn-typescript-linting`](https://github.com/MatterhornDev/learn-typescript-linting). Copy, paste and execute the following command to clone it to your machine:
+
+```bash
+git clone https://github.com/MatterhornDev/learn-typescript-linting.git
+cd learn-typescript-linting
+git checkout init
+npm install
+```
+
+The repository comes with multiple branches for different points in the guide.
+- `init`: a baseline repo without ESLint installed so you can follow along (section [2](#2-adding-eslint))
+- `master`: a complete example of TypeScript with ESLint (section [2](#2-adding-eslint))
+- `unused-variable`: an example of a common TypeScript + ESLint error (section [2.9](#29-fixing-unused-variable-definition-error-from-type-import))
+- `standard-style`: a complete example of with Standard (section [3](#3-adding-standard-style-formatter))
+- `prettier-style`: a complete example of with Prettier (section [4](#4-adding-prettier-style-formatter))
+
+The project comes with a single developer dependency, `typescript`, and two npm scripts, `compile` and `start`. The `compile` command is `tsc -p tsconfig.json`. The project is configured for `es5` in `strict` mode and includes all `.ts` files under the `src` directory. The compiled output can be found in the `lib` directory. The `start` command runs the compiled `.js` output via `node lib/index.js`. Try them out by running:
+
+```bash
+npm run compile
+npm run start
+```
+
+It is recommended you do not modify the `.ts` files as they are specifically set up to show off unique aspects of linting typescript projects.
+
+## 2 Adding ESLint
+[`[toc]`](#table-of-contents)
+
+Install `eslint`, `@typescript-eslint/eslint-plugin`, `@typescript-eslint/parser` as developer dependencies. Initialize an empty eslint configuration file. I prefer to use `.json` for configuration files.
+
+```bash
+npm i -D eslint @typescript-eslint/{eslint-plugin,parser}
```

Comment:
In powershell and cmd, you have to expand the command:
```
npm i -D eslint @typescript-eslint/eslint-plugin @typescript-eslint/parser
```

Reply:
Ahh classic Windows. Do you happen to have Git Bash installed? If so could you try the commands there instead?

Reply:
Yeah I do but my computer defaults to powershell so it's something worth mentioning

Reply:
But yes it works in git bash